elasticsearch-transport/lib/elasticsearch/transport/transport/http/manticore.rb

```diff
@@ -43,97 +46,156 @@ module HTTP
         # @see Transport::Base
         #
         class Manticore
+          attr_reader :pool, :adapter, :options
           include Base
 
           def initialize(arguments={}, &block)
-            @manticore = build_client(arguments[:options] || {})
-            super(arguments, &block)
-          end
-
-          # Should just be run once at startup
-          def build_client(options={})
-            client_options = options[:transport_options] || {}
-            client_options[:ssl] = options[:ssl] || {}
+            @options = arguments[:options] || {}
+            @logger = options[:logger]
+            @adapter = Adapter.new(logger, options)
+            @healthcheck_path = options[:healthcheck_path] || "/"
+            # TODO handle HTTPS
+            normalized_hosts = arguments[:hosts].map {|h| normalize_host(h)}
+            @pool = Manticore::Pool.new(logger, @adapter, @healthcheck_path, normalized_hosts)
+            @protocol    = options[:protocol] || DEFAULT_PROTOCOL
+            @serializer = options[:serializer] || ( options[:serializer_class] ? options[:serializer_class].new(self) : DEFAULT_SERIALIZER_CLASS.new(self) )
+            @max_retries     = options[:retry_on_failure].is_a?(Fixnum)   ? options[:retry_on_failure]   : DEFAULT_MAX_RETRIES
+            @retry_on_status = Array(options[:retry_on_status]).map { |d| d.to_i }
 
-            @manticore = ::Manticore::Client.new(client_options)
+            setup_sniffing!
           end
 
-          # Performs the request by invoking {Transport::Base#perform_request} with a block.
-          #
-          # @return [Response]
-          # @see    Transport::Base#perform_request
-          #
-          def perform_request(method, path, params={}, body=nil)
-            super do |connection, url|
-              params[:body] = __convert_to_json(body) if body
-              params = params.merge @request_options
-              case method
-              when "GET"
-                resp = connection.connection.get(url, params)
-              when "HEAD"
-                resp = connection.connection.head(url, params)
-              when "PUT"
-                resp = connection.connection.put(url, params)
-              when "POST"
-                resp = connection.connection.post(url, params)
-              when "DELETE"
-                resp = connection.connection.delete(url, params)
+          def normalize_host(host)
+            case host
+              when URI
```

Comment:
indentation detail: cases should be have all clauses aligned: case,when, else, end